### PR TITLE
Closes #9211: Improve applinks non user gesture requests

### LINKED ISSUE #9211: Improve applinks non user gesture requests
Currently in applinks non user gesture requests use cases are not clearly defined.  Clarify handling to non user gesture request.

### PR DESCRIPTION
**Pull request by rocketsroger**
### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

File: components/feature/app-links/src/test/java/mozilla/components/feature/app/links/AppLinksInterceptorTest.kt [hunk 1]
```diff
@@ -154,6 +154,12 @@ class AppLinksInterceptorTest {
         assertEquals(null, response)
     }
 
+    @Test
+    fun `request is not intercepted if not user gesture but is a subframe request`() {
```
Comment (csadilek) [inline components/feature/app-links/src/test/java/mozilla/components/feature/app/links/AppLinksInterceptorTest.kt:158]: nit: can we reword this to "request is not intercepted if a subframe request and not triggered by user"?

File: components/feature/app-links/src/main/java/mozilla/components/feature/app/links/AppLinksInterceptor.kt [hunk 2]
```diff
@@ -75,6 +75,7 @@ class AppLinksInterceptor(
 
         val doNotIntercept = when {
             uriScheme == null -> true
+            (!hasUserGesture && isSubframeRequest) -> true
```
Comment (csadilek) [inline components/feature/app-links/src/main/java/mozilla/components/feature/app/links/AppLinksInterceptor.kt:78]: nit: Let's add a comment here to explain what this does: // We don't want to go to an external app for subframe requests not triggered by the user or similar.

**Review by csadilek — APPROVED** [review]
2 nits below to clarify this a bit more, otherwise 🚢

File: components/feature/app-links/src/test/java/mozilla/components/feature/app/links/AppLinksInterceptorTest.kt (hunk 1 above)
Comment (rocketsroger) [reply components/feature/app-links/src/test/java/mozilla/components/feature/app/links/AppLinksInterceptorTest.kt:158]: Sure, I'll make the change

File: components/feature/app-links/src/main/java/mozilla/components/feature/app/links/AppLinksInterceptor.kt (hunk 2 above)
Comment (rocketsroger) [reply components/feature/app-links/src/main/java/mozilla/components/feature/app/links/AppLinksInterceptor.kt:78]: Agreed.  Thanks,
